_posts/2019-04-24-server-rendering-responsively.md

```diff
@@ -0,0 +1,199 @@
+---
+layout: epic
+title: "Server-Rendering A Responsive Artsy.Net"
+date: "2019-04-22"
+author: [steve-hicks, eloy]
+categories: [react, html, web]
+---
+
+At Artsy, we use server-side rendering (SSR) to deliver every page you hit on [artsy.net](https://artsy.net). We
+decided on using SSR for many reasons, amongst them performance. (TODO: Maybe more about why we chose SSR, but I
+don't know much about that history.)
+
+(TODO: Maybe a bit more about SSR & `hydrate`, or just link to those docs.)
+
+One challenge we've run into with SSR is rendering markup and styles from the server that the browser doesn't have
+to throw away. When your browser finishes loading our JavaScript & hydrates our React app, our app suddenly has
+more information about your browser dimensions than we had in the server request. React will go through the
+reconciliation process with this new information, to determine which DOM elements need to be updated. If there are
+significant differences between the markup our server gave you and the component tree our React app is now
+rendering, there's noticeable churn in the DOM.
```

Comment:
> there's noticeable churn in the DOM.

I get the idea of what this means, but we could follow it up with a clarifying sentence to repeat the point and help readers understand the practical implications. Something like:

> there's noticeable churn in the DOM. The time-to-interaction is increased as React goes through an additional, expensive render pass.

Or something like that.

Reply:
Based on @damassi's comments of a similar nature, I think there's definitely some work I have to do to explain this better.